src/products/dto/product.dto.ts

```diff
@@ -0,0 +1,16 @@
+import { IsNotEmpty, IsNumber, IsString } from 'class-validator';
+import { StatusEnum } from '../types/status-enum';
+
+export class CreateProductDto {
+  id: string;
+  @IsString()
+  @IsNotEmpty()
+  name: string;
+  @IsNumber()
+  @IsNotEmpty()
+  price: number;
+  @IsNumber()
+  @IsNotEmpty()
+  stock: number;
+  status: StatusEnum;
```

Comment:
You can validate the property with this decorator.
```suggestion
  @IsEnum(StatusEnum, {
    message: `$property must be one of these: ${Object.keys(StatusEnum)}`,
  })
  status: StatusEnum;
```